lib/oauth2-client.js

```diff
@@ -99,24 +83,23 @@ async function _getUncachedAccessToken({
  * @param {string} options.token_endpoint - The URL to call.
  * @param {string} options.grant_type - The grant type.
  * @param {string} options.scope - The scope.
- * @param {object} options.httpsAgent - A NodeJS HTTPS Agent (`https.Agent`).
  * @param {number} options.maxRetries - The maximum number of times to retry
  *  the request.
  *
  * @returns {string} The access token.
  */
-async function getNewAccessToken({
-  client_id, client_secret, token_endpoint, grant_type, scope, httpsAgent,
-  maxRetries = 3}) {
+export async function getNewAccessToken({
+  client_id, client_secret, token_endpoint, grant_type, scope, maxRetries = 3
+}) {
   const body = new URLSearchParams({
     client_id, client_secret, grant_type, scope
   });
 
   for(; maxRetries >= 0; --maxRetries) {
     const access_token = await _requestAccessToken(
-      {url: token_endpoint, body, httpsAgent, maxRetries});
+      {url: token_endpoint, body, maxRetries});
     if(access_token) {
-      return {client_id, client_secret, access_token};
+      return {client_id, token_endpoint, access_token};
     }
   }
   throw new BedrockError(
```

Comment:
I think we need more details here than 'Service Unavailable', for our own debugging purposes.
At very least, we should say something like "Could not get access token from service `${token_endpoint}` after X retries."

Reply:
Also, let's double-check that this bedrock error is logged to the `logger` we use elsewhere in the code.

Reply:
@dmitrizagidulin There are multiple errors that are sent to `logger` throughout the process, including the initial error when trying to get an access token, and any subsequent errors on retries, and finally what the unrecoverable error was (if there was one). What other information should be logged?

Reply:
Ok, excellent (re logger), was just checking to make sure.

I do think the error itself should have more information, though (since that's what's going to be returned to the calling app). So, for example, consider the case where it's MRGAS that runs into a token access error.

Our call chain is --

POS implementer `-->` Router  `-->` MRGAS ` (tries to renew access token from) -->` Admin.

So, we know that we don't want a lot of detail to return to the Point of Sale client.
But it does make sense (for traceability) to be able to tell from a Router error log, that the 503 Service Unavailable error it received from MRGAS was caused by a it not being able to get a token after 3 retries.

@mattcollier, @dlongley - what do you think?

Reply:
+1 to minimal return to POS vendor and +1 to logging usable error that does not reveal any secrets into the the application logs.

Reply:
@dmitrizagidulin I added "name" (which comes from the `config.authorization`) and specifies the service, so the error in the logger would be `logger.debug(`Failed to acquire access token for "${name}".`, {e});`

Is that enough, or should the 503 error be changed as well?

Reply:
Yeah, lets add details to the 503 error as well.